File: packages/engine/src/cpus/6502/instructions/tax.js
import { CycleType } from '..'

/**
 * Transfer Accumulator to X
 *
 * X = A
 *
 * @example
 * MODE          SYNTAX        HEX LEN TIM
 * Implied       TAX           $AA  1   2
 */
export function* tax() {
  this.x = this.acc
  this.zero = this.x === 0
  this.negative = (this.x & 0x80) > 0
  yield CycleType.GET
}

File: packages/engine/src/cores/famibob/cpu/instructions/tax.zig
const std = @import("std");
const CPU = @import("../cpu.zig").CPU;
const Addressing = @import("../opcode.zig").Addressing;

// Transfer Accumulator to X
//
// X = A
//
// MODE          SYNTAX        HEX LEN TIM
// Implied       TAX           $AA  1   2

pub fn tax(self: *CPU, addressing: Addressing) void {
    _ = addressing;
    self.x = self.acc;
    self.z = self.x == 0;
    self.n = (self.x & 0x80) > 0;
}
